CRM/Core/BAO/Email.php

```diff
@@ -288,9 +288,14 @@ public static function holdEmail(array &$email) {
    */
   public static function domainEmails() {
     $domainEmails = [];
-    $domainFrom = (array) CRM_Core_OptionGroup::values('from_email_address');
-    foreach (array_keys($domainFrom) as $k) {
-      $domainEmail = $domainFrom[$k];
+    $domainFrom = \Civi\Api4\DomainEmailAddress::get(FALSE)
+      ->addSelect('name', 'email')
+      ->addWhere('domain_id', '=', 'current_domain')
+      ->addWhere('is_active', '=', TRUE)
+      ->addOrderBy('name')
+      ->execute();
+    foreach ($domainFrom as $address) {
+      $domainEmail = CRM_Utils_Mail::formatFromAddress($address);
       $domainEmails[$domainEmail] = htmlspecialchars($domainEmail);
```

Comment:
I wouldn't expect `htmlspecialchars` here. I would expect this to return the emails as-is, and then it's the template's job to escape them. It seems this was already here, so maybe this should remain in place for compatibility reasons.